Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5046,7 +5046,11 @@ private void PrioritizeAndSchedule(LoggingContext loggingContext, IEnumerable<No
                                 }
                             }
 
-                            pipRuntimeInfo.Priority = (criticalPath < 0 || criticalPath > MaxInitialPipPriority) ? MaxInitialPipPriority : unchecked((int)criticalPath);

```

Comment:
>pipRuntimeInfo [](start = 28, length = 14)

Signing off, but please have Kevin, Lance or Semih double-check this logic. how priority integer mixes with the critical path weight.